packages/utils/src/math/calculateApy.ts

```
@@ -0,0 +1,49 @@
+  if (Math.abs(ratePerSecond) < 0.0001) {
+    // Simple linear approximation for small rates
+    const apy = ratePerSecond * secondsInYear * 100;
+    return apy;
+  }
+
+  // For larger rates, use the compound interest formula
+  // APY = (1 + r)^n - 1
+  try {
+    const apy = (Math.pow(1 + ratePerSecond, secondsInYear) - 1) * 100;
+    // Safeguard against unrealistic values
+    if (!isFinite(apy) || apy > 1000) {
```

Biome (1.9.4)

[error] isFinite is unsafe. It attempts a type coercion. Use Number.isFinite instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isFinite instead.


(lint/suspicious/noGlobalIsFinite)